supplier/tests/test_supplier.py

```diff
@@ -0,0 +1,64 @@
+import pytest
+from conftest import Fields, TestFields
+from django.core.exceptions import FieldDoesNotExist, ValidationError
+from supplier.models import Supplier
+
+
+class TestSupplierModel:
+    @pytest.mark.django_db()
+    def test_save_supplier(self, supplier0):
+        assert supplier0 not in Supplier.objects.all()
+        Supplier.save_supplier(supplier0)
+        assert supplier0 in Supplier.objects.all()
+
+    @pytest.mark.django_db()
+    def test_delete_supplier(self, saved_supplier0):
+        assert saved_supplier0 in Supplier.objects.all()
+        Supplier.delete_supplier(saved_supplier0)
+        assert saved_supplier0 not in Supplier.objects.all()
+
+    @pytest.mark.django_db()
+    def test_delete_non_exists_supplier(self, supplier0):
+        assert supplier0 not in Supplier.objects.all()
+        Supplier.save_supplier(supplier0)
+        assert supplier0 in Supplier.objects.all()
+        Supplier.delete_supplier(supplier0)
+        assert supplier0 not in Supplier.objects.all()
+
+    @pytest.mark.django_db()
+    def test_fail_to_add_supplier(self, supplier0):
+        with pytest.raises(ValidationError):
+            supplier0.user_name = ''
+            Supplier.save_supplier(supplier0)
+
+    @pytest.mark.django_db()
+    def test_filter_by_field(self, saved_supplier0):
+        """Test - checks the Supplier method filter by field.
+
+        The test checks whether the fixtured Supplier - saved_supplier0 -
+        can be found in the returned query set of each attribute and
+        attribute value given to the method "filter by field".
+
+        Args:
+            saved_supplier0: fixture from module conftest.py.
+
+        Returns:
+            None.
+        """
+        supplier_fields = [field.name for field in Supplier._meta.get_fields()]
```

Comment:
Do we actually need the list comprehension here? Looking at the documentation, `get_fields()` should return a tuple, on which you can use the `in` operator directly.
```suggestion
        supplier_fields = Supplier._meta.get_fields()
```
If not, then maybe:
```suggestion
        supplier_fields = list(Supplier._meta.get_fields())
```